Gems/PhysX/Code/Source/EditorHeightfieldColliderComponent.cpp

```diff
@@ -383,6 +439,11 @@ namespace PhysX
 
     // SimulatedBodyComponentRequestsBus
     AzPhysics::SimulatedBody* EditorHeightfieldColliderComponent::GetSimulatedBody()
+    {
+        return const_cast<const EditorHeightfieldColliderComponent*>(this)->GetSimulatedBody();
+    }
```

Comment:
It might be worth updating `GetSimulatedBody() const` to return a `const` pointer to `AzPhysics::SimulatedBody*` (and then do the below approach recommended by Scott Meyers)

```
struct C {
  const char & get() const {
    return c;
  }
  char & get() {
    return const_cast<char &>(static_cast<const C &>(*this).get());
  }
  char c;
};
```

ref: https://stackoverflow.com/questions/123758/how-do-i-remove-code-duplication-between-similar-const-and-non-const-member-func

Reply:
Yep, made it to use `std::as_const` as the StackOverflow page suggests. 
Interestingly there's no single use of it anywhere else in the codebase..

Reply:
Yep, added `as_const` into `AZStd` and using it now. thanks!

Reply:
Perfect thank you for adding that! 🙂